imap_processing/ultra/l1b/ultra_l1b_extended.py

```diff
@@ -111,6 +123,115 @@ def get_front_y_position(start_type: ndarray, yb: ndarray) -> tuple[ndarray, nda
     return np.array(d), np.array(yf)
 
 
+def get_ph_tof_and_back_positions(
+    de_dataset: xarray.Dataset, xf: np.ndarray, sensor: str
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Calculate back xb, yb position and tof.
+
+    An incoming particle may trigger pulses from one of the stop anodes.
+    If so, four pulses are produced, one each from the north, south,
+    east, and west sides.
+
+    The Time Of Flight (tof) and the position of the particle at the
+    back of the sensor are measured using the timing of the pulses.
+    Further description is available on pages 32-33 of
+    IMAP-Ultra Flight Software Specification document
+    (7523-9009_Rev_-.pdf).
+
+    Parameters
+    ----------
+    de_dataset : xarray.Dataset
+        Data in xarray format.
+    xf : np.array
+        X front position in (hundredths of a millimeter).
+    sensor : str
+        Sensor name.
+
+    Returns
+    -------
+    tof : np.array
+        Time of flight (tenths of a nanosecond).
+    t2 : np.array
+        Particle time of flight from start to stop (tenths of a nanosecond).
+    xb : np.array
+        Back positions in x direction (hundredths of a millimeter).
+    yb : np.array
+        Back positions in y direction (hundredths of a millimeter).
+    """
+    indices = np.where(
```

Comment:
numpy recommends using np.nonzero here.
```suggestion
    indices = np.nonzero(
```

Reply:
Agreed with nonzero instead of where.

But, I'll leave another suggestion as well, which is to filter the dataset to create a subset dataset and work on that.

```python
de_filtered = de_dataset.isel(STOP_TYPE=[StopType.Top.value, StopType.Bottom.value])
```

Then work on  `de_filtered` below without needing to worry about the indices because it is already subset.

```python
sp_n_norm = get_norm(de_dataset["STOP_NORTH_TDC"], "SpN", sensor)
# or add this into the dataset as a new variable (maybe keep the same variable name and it has just been normalized?)
de_filtered["SP_N_NORM"] = get_norm(de_filtered["STOP_NORTH_TDC"], "SpN", sensor)
```